lib/BattleHex.cpp

```diff
@@ -102,58 +102,54 @@ BattleHex& BattleHex::moveInDir(EDir dir, bool hasToBeValid)
 	default:
 		throw std::runtime_error("Disaster: wrong direction in BattleHex::operator+=!\n");
 		break;
-}
+	}
 	return *this;
 }
 
 BattleHex &BattleHex::operator+=(BattleHex::EDir dir)
 {
-	return moveInDir(dir);
+	return moveInDirection(dir);
 }
 
-BattleHex BattleHex::movedInDir(BattleHex::EDir dir, bool hasToBeValid) const
+BattleHex BattleHex::cloneInDirection(BattleHex::EDir dir, bool hasToBeValid) const
 {
-	BattleHex result(*this);
-	result.moveInDir(dir, hasToBeValid);
+	BattleHex result(hex);
+	result.moveInDirection(dir, hasToBeValid);
 	return result;
 }
 
 BattleHex BattleHex::operator+(BattleHex::EDir dir) const
 {
-	return movedInDir(dir);
+	return cloneInDirection(dir);
 }
 
 std::vector<BattleHex> BattleHex::neighbouringTiles() const
 {
 	std::vector<BattleHex> ret;
-	const int WN = GameConstants::BFIELD_WIDTH;
-	// H3 order : TR, R, BR, BL, L, TL (T = top, B = bottom ...)
-
-	checkAndPush(hex - ( (hex/WN)%2 ? WN+1 : WN ), ret); // 1
-	checkAndPush(hex + 1, ret); // 2
-	checkAndPush(hex + ( (hex/WN)%2 ? WN : WN+1 ), ret); // 3
-	checkAndPush(hex + ( (hex/WN)%2 ? WN-1 : WN ), ret); // 4
-	checkAndPush(hex - 1, ret); // 5
-	checkAndPush(hex - ( (hex/WN)%2 ? WN : WN-1 ), ret); // 6
-
+	checkAndPush(cloneInDirection(TOP_LEFT, true), ret);

```

Comment:
// H3 order : TR, R, BR, BL, L, TL (T = top, B = bottom ...)
TOP_LEFT should be last